src/lib/support/BufferReader.h

```diff
@@ -156,6 +156,23 @@ class Reader
     template <typename T>
     void RawRead(T * retval);
 
+    /**
+     * Advance the Reader forward by the specified number of octets.
+     *
+     * @param len The number of octets to skip.
+     *
+     * @note If the len argument is greater than the number of available octets
+     *       remaining, the Reader will advance to the end of the buffer
+     *       without entering a failed-status state.
+     */
+    Reader & Skip(uint16_t len)
```

Comment:
Replace uint16_t with size_t please

Reply:
mAvailable is currently uint16_t.

if you want, we can define a size_type in the class, however I believe uint16_t is correct here.

Reply:
The signature of the constructor is `Reader(const uint8_t * buffer, uint16_t buf_len)` - does that `uint16_t` also need to be changed to `size_t`